lib/travis/build/addons/transifex.rb

```diff
@@ -0,0 +1,134 @@
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class Transifex < Base
+        CLIENT_VERSION = '==0.10'
+        SUPER_USER_SAFE = true
+        DEFAULTS = {
+          project: 'default',
+          host: 'https://www.transifex.com',
+          debug: false,
+          traceback: false,
+          pull: {
+            all: true,
+            force: true,
+            minimum_perc: 75,
+          }
+        }.freeze
+
+        attr_reader :validation_errors
+
+        def before_script
+          validate! && run_before
+        end
+
+        def after_script
+          run_after if valid?
+        end
+
+        private
+
+          def run_before
+            install
+            configure
+            propose
+          end
+
+          def run_after
+            update
+          end
+
+          def warn_invalid
+            validation_errors.each { |error| sh.echo error, ansi: :red }
+          end
+
+          def install
+            sh.echo 'Installing Transifex Client (beta)', ansi: :yellow
+            sh.cmd "pip install --user transifex-client#{CLIENT_VERSION}", echo: true
+            sh.export 'PATH', '$HOME/.local/bin:$PATH', echo: true
+          end
+
+          def configure
+            sh.cmd "tx init #{tx_init_options}", echo: true
+            sh.cmd "tx set #{tx_set_source_options}", echo: true
+          end
+
+          def propose
+            # TODO: propose translation update
+            sh.cmd "tx #{tx_global_options} pull #{tx_first_pull_options}", echo: true
+            sh.cmd "tx #{tx_global_options} pull #{tx_second_pull_options}", echo: true
+          end
+
+          def update
+            # TODO: update upstream translations
+            sh.cmd "tx #{tx_global_options} push", echo: true
+          end
+
+          def valid?
+            !!@valid
+          end
+
+          def validate!
+            # TODO: define what constitutes a valid configuration, such as relationship to pull requests, disabled
+            # branches, etc.
+            @valid = true
+            @validation_errors = []
+            warn_invalid unless @valid
+            @valid
+          end
+
+          def tx_config
+            @tx_config ||= config[:transifex].deep_symbolize_keys.reverse_merge(DEFAULTS)
+          end
+
+          def tx_init_options
+            @tx_init_options ||= [].tap do |o|
+              o << "--host=#{tx_config[:host]}"
+            end.join(' ')
+          end
+
+          def tx_set_source_options
+            @tx_set_source_options ||= [].tap do |o|
```

Comment:
Good stuff. We need to keep in mind that not all frameworks have the same file structure. So this can be an optional option, but people can actually have this configuration stored directly in the repo. Some resources might have multiple resource files too. Ex. https://github.com/bikalabs/Bika-LIMS/blob/master/bika/lims/locales/.tx/config

More reference: http://docs.transifex.com/developer/client/config